### PR TITLE
Clean up various things

### PR DESCRIPTION
**Pull request by cliffkoh**
It will be quite the effort to comprehensively clean up everything but this is a start.

### CODE REVIEW

File: src/client/js/oddone.local.ts [hunk 1]
```diff
@@ -33,11 +33,11 @@ $(() => {
         });
 
     _mplib.MultiplayR.Host('theoddone',
-                           transport,
-                           document.getElementById('host-container'),
-                           (res) => {
-                               _mplib.messages.checkReturnMessage(res, 'roomId');
-                               setupJoin(res.message);
-                           });
+        transport,
```
Comment (zichun) [inline src/client/js/oddone.local.ts]: imo this makes for terrible reading. arguments should always be aligned with the indentation of the previous argument. if you prefer such a style, then the first argument should start on its own line

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Place your settings in this file to overwrite default and user settings.
```
Comment (cliffkoh) [inline .vscode/settings.json]: We've tried some which has these things checked in and some which have not and overall found that checking in works pretty well and also helps ensure everyone is coding in a consistent way (e.g. number of spaces for each tab, if 2 was adopted instead. 2 seems to be pretty common/standard for web projects in particular).

File: src/rules/gameshell/gameshell.tsx [hunk 3]
```diff
@@ -29,24 +31,24 @@ export const Shell: GameRuleInterface = {
                 const header = React.createElement(Shell.views['HostShell-Main-Head'], this.props);
                 const body = React.createElement(Shell.views['HostShell-Main-Body'], this.props);
 
-                return React.DOM.div({id: 'shell-main'},
-                                     header,
-                                     body);
+                return React.DOM.div({ id: 'shell-main' },
```
Comment (zichun) [inline src/rules/gameshell/gameshell.tsx]: the indentation here needs to be fixed (first argument on a new line)

File: src/rules/gameshell/gameshell.tsx [hunk 4]
```diff
@@ -29,24 +31,24 @@ export const Shell: GameRuleInterface = {
...
+                    body);
             }
         },
 
         'HostShell-Main-Head': class extends React.Component<ViewPropsInterface, {}> {
             public render() {
-                return React.DOM.div({id: 'shell-header'},
-                                     React.DOM.div({id: 'shell-room'},
-                                                   this.props.MP.roomId));
+                return React.DOM.div({ id: 'shell-header' },
```
Comment (zichun) [inline src/rules/gameshell/gameshell.tsx]: fix indentation (this is exactly why arguments should be aligned with the indentation of previous argument)

File: src/rules/gameshell/gameshell.tsx [hunk 5]
```diff
@@ -60,18 +62,23 @@ export const Shell: GameRuleInterface = {
                 });
 
                 const content = React.DOM.div({ id: 'shell-main-content' },
```
Comment (zichun) [inline src/rules/gameshell/gameshell.tsx]: fix pls

File: src/rules/gameshell/gameshell.tsx [hunk 6]
```diff
@@ -82,12 +89,12 @@ export const Shell: GameRuleInterface = {
             }
 
             public render() {
-                const tr = [React.DOM.li({onClick: this.setView, key: 'home'}, 'Home')];
+                const tr = [React.DOM.li({ onClick: this.setView, key: 'home' }, 'Home')];
                 for (let i = 0; i < this.props.links.length; i = i + 1) {
-                    tr.push(React.DOM.li({onClick: this.setView, key: ('link-' + i)}, this.props['links'][i]));
+                    tr.push(React.DOM.li({ onClick: this.setView, key: ('link-' + i) }, this.props['links'][i]));
                 }
                 return React.DOM.ul({ className: 'shell-navigation' },
```
Comment (zichun) [inline src/rules/gameshell/gameshell.tsx]: fix

File: src/server/Room.ts [hunk 7]
```diff
@@ -50,61 +54,67 @@ export class Room {
...
-                     session: Session) {
+    /**
+     * Add Client to Room
+     * @param clientId Unique Id of client
+     * @param socket The socket.io object of the new client
+     */
+    public addClient(
+        clientId: string,
+        session: Session
+    ) {
```
Comment (zichun) [inline src/server/Room.ts]: need to standardize whether the closing parenthesis and opening brace ') {' start on the new line or not

File: src/server/Room.ts [hunk 8]
```diff
@@ -50,61 +54,67 @@ export class Room {
...
         }
 
-        this.clients.push(clientId);
-        this.clientSessions[clientId] = session;
-        this.clientActiveMap[clientId] = true;
+        this._clients.push(clientId);
+        this._clientSessions[clientId] = session;
+        this._clientActiveMap[clientId] = true;
 
         this.broadcastRoomActivity(RoomMessageType.JoinRoom,
```
Comment (zichun) [inline src/server/Room.ts]: fix

File: src/server/Room.ts [hunk 9]
```diff
@@ -115,58 +125,61 @@ export class Room {
...
         }
 
-        this.clientActiveMap[clientId] = true;
-        this.clientSessions[clientId] = session;
+        this._clientActiveMap[clientId] = true;
+        this._clientSessions[clientId] = session;
 
         const broadcastType = RoomMessageType.RejoinRoom;
 
         this.broadcastRoomActivity(broadcastType,
```
Comment (zichun) [inline src/server/Room.ts]: fix

File: src/server/Room.ts [hunk 10]
```diff
@@ -115,58 +125,61 @@ export class Room {
...
         if (index === -1) {
             return false;
         }
 
-        this.clientActiveMap[clientId] = false;
-        this.clientSessions[clientId] = null;
+        this._clientActiveMap[clientId] = false;
+        this._clientSessions[clientId] = null;
 
         this.broadcastRoomActivity(RoomMessageType.LeaveRoom,
```
Comment (zichun) [inline src/server/Room.ts]: fix

File: src/server/session.ts [hunk 11]
```diff
@@ -163,9 +171,9 @@ export class Session implements ServerSessionInterface {
         }
 
         return this.room.sendMessage(toClientId,
```
Comment (zichun) [inline src/server/session.ts]: fix

File: src/server/socket.transport.ts [hunk 12]
```diff
@@ -37,61 +43,61 @@ export class SocketTransport implements ServerTransportInterface {
...
         this.socket.on('initialize',
-                       (data: any, cb: CallbackType) => {
+            (data: any, cb: CallbackType) => {
 
-                           if (this.initialized) {
-                               return returnError(cb,
-                                                  'transport has already been initialized (clientId: ' + this.session.getClientId() + ')');
-                           }
+                if (this.initialized) {
+                    return returnError(cb,
```
Comment (zichun) [inline src/server/socket.transport.ts]: fix

File: src/server/socket.transport.ts [hunk 13]
```diff
@@ -37,61 +43,61 @@ export class SocketTransport implements ServerTransportInterface {
...
-                                                  data.clientId,
-                                                  (res) => {
-                                                      if (res.success === false) {
-                                                          this.session = null;
-                                                          this.initialized = false;
-                                                      }
-                                                      return forwardReturnMessage(res, cb);
-                                                  });
-                       });
+            (data: ReconnectPacketType, cb: CallbackType) => {
```
Comment (zichun) [inline src/server/socket.transport.ts]: fix

File: src/server/socket.transport.ts [hunk 14]
```diff
@@ -37,61 +43,61 @@ export class SocketTransport implements ServerTransportInterface {
...
+                // todo: handle reconnection logic
+                if (this.session || this.initialized) {
+                    return returnError(cb,
```
Comment (zichun) [inline src/server/socket.transport.ts]: fix

File: src/server/socket.transport.ts [hunk 15]
```diff
@@ -37,61 +43,61 @@ export class SocketTransport implements ServerTransportInterface {
...
+                        'transport has already been initialized, cannot rejoin (clientId: ' +
+                        this.session.getClientId() + ')');
+                }
+
+                if (!data.roomId || !data.clientId) {
+                    return returnError(cb,
```
Comment (zichun) [inline src/server/socket.transport.ts]: fix

File: src/server/socket.transport.ts [hunk 16]
```diff
@@ -37,61 +43,61 @@ export class SocketTransport implements ServerTransportInterface {
...
+                        'invalid reconnection packet (missing data.roomId/clientId)');
+                }
+
+                this.initialized = true;
+                this.session = new Session(this);
+                this.session.reconnect(data.roomId,
```
Comment (zichun) [inline src/server/socket.transport.ts]: fix

File: src/server/socket.transport.ts [hunk 17]
```diff
@@ -37,61 +43,61 @@ export class SocketTransport implements ServerTransportInterface {
...
+                            this.session = null;
+                            this.initialized = false;
+                        }
+                        return forwardReturnMessage(res, cb);
+                    });
+            });
 
         this.socket.on('message',
-                       (packet: PacketType, cb: CallbackType) => {
+            (packet: PacketType, cb: CallbackType) => {
```
Comment (zichun) [inline src/server/socket.transport.ts]: fix

File: src/server/socket.transport.ts [hunk 18]
```diff
@@ -109,8 +115,8 @@ export class SocketTransport implements ServerTransportInterface {
         cb?: CallbackType
     ) {
         this.socket.emit('message',
-                         packet,
-                         cb);
+            packet,
```
Comment (zichun) [inline src/server/socket.transport.ts]: fix

**Pull request merged**